src/Platform/Microsoft.Testing.Platform/OutputDevice/Console/ConsoleLogger.cs

```diff
@@ -0,0 +1,702 @@
+        terminal.Append(' ');
+        AppendAssemblyResult(terminal, assemblyRun.FailedTests == 0, failedTests, warnings);
+        terminal.Append(' ');
+        AppendDuration(terminal, assemblyRun.Stopwatch.Elapsed);
+    }
+
+    private static void AppendDuration(ITerminal terminal, TimeSpan duration, bool wrapInParentheses = true, bool colorize = true)
+    {
+        if (colorize)
+        {
+            terminal.SetColor(TerminalColor.Gray);
+        }
+
+        if (wrapInParentheses)
+        {
+            terminal.Append('(');
+        }
+
+        if (duration.TotalMilliseconds < 1000)
+        {
+            terminal.Append(duration.TotalMilliseconds.ToString("F0", CultureInfo.CurrentCulture));
+            terminal.Append("ms");
+        }
+        else
+        {
+            terminal.Append(duration.TotalSeconds.ToString("F1", CultureInfo.CurrentCulture));
+            terminal.Append('s');
+        }
+
+        if (wrapInParentheses)
+        {
+            terminal.Append(')');
+        }
+
+        if (colorize)
+        {
+            terminal.ResetColor();
+        }
+    }
+
+    public void Dispose() => _consoleWithProgress.Dispose();
+
+    public void ArtifactAdded(bool outOfProcess, string? assembly, string? targetFramework, string? architecture, string? testName, string path)
+        => _artifacts.Add(new LoggerArtifact(outOfProcess, assembly, targetFramework, architecture, testName, path));
+
+    /// <summary>
+    /// Let the user know that cancellation was triggered.
+    /// </summary>
+    internal void StartCancelling()
+    {
+        _wasCancelled = true;
+        _consoleWithProgress.WriteToTerminal(terminal =>
+        {
+            terminal.AppendLine();
+            terminal.AppendLine(PlatformResources.CancellingTestSession);
+            terminal.AppendLine();
+        });
+    }
+
+    internal void WriteErrorMessage(string assembly, string? targetFramework, string? architecture, string text)
+    {
+        TestModule asm = GetOrAddAssemblyRun(assembly, targetFramework, architecture);
+        asm.AddError(text);
+
+        _consoleWithProgress.WriteToTerminal(terminal =>
+        {
+            terminal.SetColor(TerminalColor.Red);
+            terminal.AppendLine(text);
+            terminal.ResetColor();
+        });
+    }
+
+    internal void WriteWarningMessage(string assembly, string? targetFramework, string? architecture, string text)
+    {
+        TestModule asm = GetOrAddAssemblyRun(assembly, targetFramework, architecture);
+        asm.AddWarning(text);
+        _consoleWithProgress.WriteToTerminal(terminal =>
+        {
+            terminal.SetColor(TerminalColor.Yellow);
+            terminal.AppendLine(text);
+            terminal.ResetColor();
+        });
+    }
+
+    internal void WriteErrorMessage(string assembly, string? targetFramework, string? architecture, Exception exception) =>
+        WriteErrorMessage(assembly, targetFramework, architecture, exception.ToString());
```

Comment:
Nit: for consistency with the other methods

```suggestion
    internal void WriteErrorMessage(string assembly, string? targetFramework, string? architecture, Exception exception)
        => WriteErrorMessage(assembly, targetFramework, architecture, exception.ToString());
```

Reply:
did the change in place.